src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -118,6 +124,76 @@ public synchronized void close() throws IOException {
+        } finally {
+            stopForcefully();
+        }
+    }
+
+    @SuppressFBWarnings("SWL_SLEEP_WITH_LOCK_HELD")
+    private void stopGracefully() {
+        //TODO: Temporarily limiting graceful shutdown to only processes that are created using ProcessImplForWin32
+        if (!(process instanceof ProcessImplForWin32)) {
+            return;
+        }
+        int pid = process instanceof ProcessImplForWin32
+                ? ((ProcessImplForWin32) process).getPid() : Processes.newPidProcess(process).getPid();
+        boolean sentConsoleCtrlEvent = false;
+        synchronized (Kernel32.INSTANCE) {
+            Kernel32 k32 = Kernel32.INSTANCE;
+            // if the console is already attached, a second attach call will fail.
+            if (!k32.FreeConsole()) {
+                logger.error("FreeConsole error {}", k32.GetLastError());
+            }
+            if (!k32.AttachConsole(pid)) {
+                logger.error("AttachConsole error {}", k32.GetLastError());
+                // Console already attached so we cannot signal it
+                if (k32.GetLastError() == ERROR_ACCESS_DENIED) {
+                    logger.info("AttachConsole failed for PID: {}, calling FreeConsole on it", pid);
+                    // Oddly, calling FreeConsole seems to make subsequent calls work?
+                    if (!k32.FreeConsole()) {
+                        logger.error("FreeConsole error {}", k32.GetLastError());
+                    }
+                }
+                return;
+            }
+            Kernel32Ex k32Ex = Kernel32Ex.INSTANCE;
+            try {
+                if (!k32Ex.SetConsoleCtrlHandler(null, true)) {
+                    logger.error("SetConsoleCtrlHandler add error {}", k32.GetLastError());
+                    return;
+                }
+                if (k32.GenerateConsoleCtrlEvent(Wincon.CTRL_C_EVENT, 0)) {
+                    sentConsoleCtrlEvent = true;
+                } else {
+                    logger.error("GenerateConsoleCtrlEvent error {}", k32.GetLastError());
+                }
+            } finally {
+                if (!k32.FreeConsole()) {
+                    logger.error("FreeConsole error {}", k32.GetLastError());
+                }
+                // wait to ensure CtrlHandler is not enabled before the calling process gets the signal
+                try {
+                    Thread.sleep(2000);
+                } catch (InterruptedException ignored) {
+                }
+                if (!k32Ex.SetConsoleCtrlHandler(null, false)) {
+                    logger.error("SetConsoleCtrlHandler remove error {}", k32.GetLastError());
+                }
+            }
+        }
+
+        try {
+            if (sentConsoleCtrlEvent) {
+                process.waitFor(gracefulShutdownTimeout, TimeUnit.SECONDS);
+                logger.info("Process stopped gracefully: {}", pid);
+            }
+        } catch (InterruptedException ignore) {
+        }
```

Comment:
don't ignore it, throw it up or at least re-set the interrupted flag on the thread